core/bridges/orm.go

```diff
@@ -266,3 +289,28 @@ func (o *orm) FindExternalInitiatorByName(ctx context.Context, iname string) (ex
 	err = o.ds.GetContext(ctx, &exi, `SELECT * FROM external_initiators WHERE lower(name) = lower($1)`, iname)
 	return
 }
+
+func (o *orm) upsertResponse(ctx context.Context, dotId string, specId int32, response []byte, nextDelay time.Duration) {
+	sql := `INSERT INTO bridge_last_value(dot_id, spec_id, value, finished_at) 
+				VALUES($1, $2, $3, $4)
+			ON CONFLICT ON CONSTRAINT bridge_last_value_pkey
+				DO UPDATE SET value = $3, finished_at = $4;`
+
+	_, err := o.ds.ExecContext(ctx, sql, dotId, specId, response, time.Now())
+	if err != nil {
+		// TODO: log a warning and retry
+		time.Sleep(nextDelay)
```

Comment:
This can lead to resource exhaustion as implemented

Reply:
How quickly are these records being saved? Or better, how many records per 10s should we be writing to the database?

Reply:
Depends on the product, mercury will do hundreds per second or more

Reply:
If you get a persistent error e.g. some constraint is getting violated or DB is unavailable this would spawn unbounded goroutines as written